src/compiler/checker.ts

```diff
@@ -11243,9 +11243,16 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     function getTypeOfAlias(symbol: Symbol): Type {
         const links = getSymbolLinks(symbol);
         if (!links.type) {
+            if (!pushTypeResolution(symbol, TypeSystemPropertyName.Type)) {
+                return errorType;
+            }
             const targetSymbol = resolveAlias(symbol);
             const exportSymbol = symbol.declarations && getTargetOfAliasDeclaration(getDeclarationOfAliasSymbol(symbol)!, /*dontResolveAlias*/ true);
             const declaredType = firstDefined(exportSymbol?.declarations, d => isExportAssignment(d) ? tryGetTypeFromEffectiveTypeNode(d) : undefined);
+
+            if (!popTypeResolution()) {
+                links.type = errorType;

```

Comment:
the other uses of popTypeResolution either do something like `return links.type = errorType` or otherwise make sure that errorType is the type assigned to `links.type`. Right now, the next assignment, on line 11261, will overwrite `links.type` in some cases, but shouldn't.